src/model/test/ModelObject_GTest.cpp

```diff
@@ -125,3 +127,25 @@ TEST_F(ModelFixture, ModelObject_Clone_DifferentModel) {
   EXPECT_TRUE(anotherNewSurface.construction().get().cast<LayeredConstruction>().layers()
               == newSurface.construction().get().cast<LayeredConstruction>().layers());
 }
+
+TEST_F(ModelFixture, ModelObject_SetString) {
```

Comment:
This is a new test to verify the bug fix

Reply:
Merge from #4206 adds a very similar one for at Workspace level, I think both are relevant and I don't see a problem testing twice.